core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolImpl.java

```diff
@@ -562,9 +564,20 @@ public void setCreationTime(Date created) {
   }
 
   @Override
-  public int getLongestIndex() {
-    return pooledElementViews.stream().flatMap(element -> element.getIndices().stream()).mapToInt(index -> index.getSequence().length())
-        .max().orElse(0);
+  public String getLongestIndex() {
+    Map<Integer, Integer> lengths = pooledElementViews.stream()
+        .flatMap(element -> element.getIndices().stream())
+        .collect(Collectors.toMap(Index::getPosition, index -> index.getSequence().length(), Integer::max));
+    if (lengths.isEmpty()) {
+      return "0";
+    }
+    return lengths.entrySet().stream()
+        .sorted((a, b) -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)